### PR TITLE
[5.7] WhereToday feature

### PR DESCRIPTION
**Pull request by dmitryuk**
In my and some other projects code often have a lot of queries that filter entries by today condition.
This PR introduces "whereToday" feature in Query Builder.

### CODE REVIEW

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -1299,6 +1299,28 @@ public function orWhereYear($column, $operator, $value = null)
...
 
+    /**
+     * Add an "where today" statement in the query.
+     *
+     * @param  string $column
+     * @return \Illuminate\Database\Query\Builder|static
+     */
+    public function whereToday($column)
+    {
+        return $this->whereDate($column, '=', today());
```
Comment (mfn) [inline src/Illuminate/Database/Query/Builder.php]: You can't use `today()`, it's a global helper from Laravel Framework itself.

However `illuminate/database` is a self-contained code base/package which can be used outside of Laravel, thus you can't depend on this.

Also:  it misses passing the time zone, this is vital. Not everyone would require to only use the applications default timezone.

Comment (xakzona) [reply src/Illuminate/Database/Query/Builder.php]: I have changed Carbon::today() to DateTime with additional $timeZone parameter

**Comment by AbdelElrafa** [comment]
I agree with @ahinkle. This seems like it should be a scope you add per project that you need it, and not something added to the framework. Perhaps you could make a nice package to do this and include all the examples that @ahinkle gave.

File: src/Illuminate/Database/Query/Builder.php [hunk 2]
```diff
@@ -1299,6 +1299,32 @@ public function orWhereYear($column, $operator, $value = null)
...
+     * Add an "where today" statement in the query.
+     *
+     * @param  string $column
+     * @param \DateTimeZone|null $timeZone
+     * @return \Illuminate\Database\Query\Builder|static
+     * @throws \Exception
+     */
+    public function whereToday($column, ?\DateTimeZone $timeZone = null)
+    {
+        return $this->whereDate($column, '=', new \DateTime('now', $timeZone));
```
Comment (mfn) [inline src/Illuminate/Database/Query/Builder.php]: Just FTR (not that it matters anymore), it should have been `\Illuminate\Support\Carbon` => this is the default the framework uses (it's defined in the `illuminate/support` package which other, framework independent ones, usually depend one).

**Pull request closed without merging**